src/main/java/com/aws/greengrass/clientdevices/auth/configuration/RuntimeConfiguration.java

```diff
@@ -66,4 +92,116 @@ public void updateCAPassphrase(String passphrase) {
         caPassphrase.withValue(passphrase);
     }
 
+    /**
+     * Retrieve a Thing.
+     *
+     * @param thingName ThingName
+     * @return Optional of ThingV1 DTO, else empty optional
+     */
+    public Optional<ThingV1> getThingV1(String thingName) {
+        Topics v1ThingTopics = config.findTopics(THINGS_KEY, THINGS_V1_KEY, thingName);
+
+        if (v1ThingTopics == null) {
+            return Optional.empty();
+        }
+
+        Topics certTopics = v1ThingTopics.findTopics(THINGS_CERTIFICATES_KEY);
+        if (certTopics == null || certTopics.isEmpty()) {
+            return Optional.of(new ThingV1(thingName, Collections.emptyMap()));
+        }
+        Map<String, Long> certMap = new HashMap<>();
+        certTopics.forEach(node -> {
+            certMap.put(node.getName(), Coerce.toLong(node));
+        });
+
+        return Optional.of(new ThingV1(thingName, certMap));
+    }
+
+    /**
+     * Store a Thing in the Runtime Configuration.
+     *
+     * @param thing Thing DTO
+     */
+    public void putThing(@NonNull ThingV1 thing) {
+        Topics v1ThingTopics = getOrRepairTopics(config, THINGS_KEY, THINGS_V1_KEY, thing.getThingName());
+        Map<String, Object> certMap = new HashMap<>(thing.getCertificates());
```

Comment:
do we need this? Can't you just pass in the `thing.getCertificates()` directly to `replaceAndWait`?